feat(*): support 1.1 kinds

technosophos opened the pull request:
This adds support for several 1.1 kinds, and also generalizes many of
the data structures.

Addresses #253, #254, #255, #256, and partially #194.

arschles commented on action/install.go:
```diff
@@ -86,44 +92,27 @@ func isSamePath(src, dst string) (bool, error) {
-		if err := marshalAndCreate(o, namespace, dryRun); err != nil {
-			return err
+
+	log.Warn("Matt forgot to write the code to modify the versioned object.")
```
is this log line intentional?

technosophos replied:
Yeah, this is still a work in progress, and I have some stuff in there to shame me into remembering to do things. ;-)

arschles replied:
:+1:

arschles commented on chart/chart.go:
```diff
@@ -61,101 +67,101 @@ func Load(chart string) (*Chart, error) {
-		}
+		c.Kind[m.Kind] = append(c.Kind[m.Kind], m)
+
+		//vo := m.VersionedObject
```
remove this commented code?

technosophos commented:
The 1.1 support is now ready for review. This will impact outstanding PRs that have Kinds (e.g. #281 ). But it should impact them in a Good Way (TM).

Some things to note:

- The only place that now knows anything about decoding particular Kinds is package `codec`
- I added testdata for all of the new v1beta1 kinds in v1.1.1
- As part of this refactor, I also fixed the way annotations were being named, and stored the annotations that #194 requested. This seemed like the right place to add those because I already had to rewrite the existing annotations code.

bacongobbler commented on action/install.go:
```diff
@@ -13,18 +13,24 @@ import (
+// Unknown manifest types (those not explicitly referenced in this list) will
+// be uninstalled before any of these, since we know that none of the core
+// types depend on non-core types.
+var UninstallOrder = []string{"Service", "Pod", "ReplicationController", "DaemonSet", "Secret", "PersistentVolume", "ServiceAccount", "Ingress", "Job"}
```
namespaces are never uninstalled?

technosophos replied:
I can't decide what to do about namespaces. They feel so snowflake-like. It felt dangerous to remove an entire namespace, but I don't actually have a use case that suggests we shouldn't. Maybe I'll put them in for now, and if that causes problems I'll take them back out.

technosophos replied:
@bacongobbler that was a good catch. I had forgotten that if the type isn't explicitly set in this list, it gets deleted *before* all the declared ones. That would have had some nasty side-effects.

bacongobbler replied:
Ah yes, it would be uninstalled as part of the `UnknownTypes` structure. Good point.

adamreese commented:
k8s object type switches deleted :clap:

technosophos commented:
Fixed @bacongobbler 's excellent catch on namespaces.
